cppcache/src/TcrEndpoint.cpp

```diff
@@ -504,14 +504,15 @@ void TcrEndpoint::pingServer(ThinClientPoolDM* poolDM) {
   }
 
   if (!m_msgSent && !m_pingSent) {
-    TcrMessagePing* pingMsg = TcrMessage::getPingMessage(m_cacheImpl);
+    TcrMessagePing pingMsg(new DataOutput(m_cacheImpl->createDataOutput()),
+                           true);
```

Comment:
If this value is always `true` then let's just roll it into the constructor initializer and remove from parameters.